packages/fxa-settings/src/pages/Signin/SigninRecoveryChoice/container.tsx

```diff
@@ -0,0 +1,127 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { useState, useEffect } from 'react';
+import { RouteComponentProps, useLocation } from '@reach/router';
+import SigninRecoveryChoice from '.';
+import { useAuthClient } from '../../../models';
+import { useNavigateWithQuery } from '../../../lib/hooks/useNavigateWithQuery';
+import { SigninLocationState } from '../interfaces';
+import { getSigninState } from '../utils';
+import LoadingSpinner from 'fxa-react/components/LoadingSpinner';
+import {
+  AuthUiErrorNos,
+  AuthUiErrors,
+} from '../../../lib/auth-errors/auth-errors';
+
+export const SigninRecoveryChoiceContainer = (_: RouteComponentProps) => {
+  const authClient = useAuthClient();
+  const location = useLocation() as ReturnType<typeof useLocation> & {
+    state: SigninLocationState;
+  };
+  const navigateWithQuery = useNavigateWithQuery();
+  const signinState = getSigninState(location.state);
+
+  const [numBackupCodes, setNumBackupCodes] = useState<number>(0);
+  const [lastFourPhoneDigits, setLastFourPhoneDigits] = useState<string>('');
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    if (!signinState || !signinState.sessionToken) {
+      navigateWithQuery('/signin');
+      return;
+    }
+
+    const fetchData = async () => {
+      try {
+        const { count } = await authClient.getRecoveryCodesExist(
+          signinState.sessionToken
+        );
+        count && setNumBackupCodes(count);
+
+        const { phoneNumber } = await authClient.recoveryPhoneGet(
+          signinState.sessionToken
+        );
+        // TODO verify that recoveryPhoneGet returns a masked phone number (last four digits only)
+        phoneNumber && setLastFourPhoneDigits(phoneNumber.slice(-4));
```

Comment:
You might could create a little helper function for this: https://github.com/mozilla/fxa/blob/995592429c351d2ba9393d486db92e33f5b3e8c4/packages/fxa-settings/src/components/Settings/SubRow/index.tsx#L240-L241

See:
* https://github.com/mozilla/fxa/blob/995592429c351d2ba9393d486db92e33f5b3e8c4/libs/accounts/recovery-phone/src/lib/recovery-phone.service.ts#L192-L193
* https://github.com/mozilla/fxa/blob/995592429c351d2ba9393d486db92e33f5b3e8c4/libs/accounts/recovery-phone/src/lib/recovery-phone.service.ts#L259-L260

Reply:
I added a note in `FXA-11044`